packages/image_picker/image_picker_ios/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -119,7 +119,12 @@ - (void)launchPHPickerWithContext:(nonnull FLTImagePickerMethodCallContext *)con
   _pickerViewController.presentationController.delegate = self;
   self.callContext = context;
 
-  [self checkPhotoAuthorizationForAccessLevel];
+  BOOL requestFullMetadata = context.requestFullMetadata;
```

Comment:
Nit: there's no value to this local variable given how short and readable the right hand side is; you can just inline `context.requestFullMetadata` into the `if` condition.